diff --git a/core/lib/Drupal/Core/Language/LanguageManager.php b/core/lib/Drupal/Core/Language/LanguageManager.php
index c0177df1..65d2db46 100644
--- a/core/lib/Drupal/Core/Language/LanguageManager.php
+++ b/core/lib/Drupal/Core/Language/LanguageManager.php
@@ -292,6 +292,7 @@ public static function getStandardLanguageList() {
       'mr' => ['Marathi', 'मराठी'],
       'ms' => ['Bahasa Malaysia', 'بهاس ملايو'],
       'my' => ['Burmese', 'ဗမာစကား'],
+      'myn' => ['Mayan', 'Kaqchikel'],
       'ne' => ['Nepali', 'नेपाली'],
       'nl' => ['Dutch', 'Nederlands'],
       'nb' => ['Norwegian Bokmål', 'Norsk, bokmål'],
@@ -318,6 +319,7 @@ public static function getStandardLanguageList() {
       'th' => ['Thai', 'ภาษาไทย'],
       'tr' => ['Turkish', 'Türkçe'],
       'tyv' => ['Tuvan', 'Тыва дыл'],
+      'tir' => ['Tigrinya', 'ትግርኛ'],
       'ug' => ['Uyghur', /* Left-to-right marker "‭" */ 'ئۇيغۇرچە', LanguageInterface::DIRECTION_RTL],
       'uk' => ['Ukrainian', 'Українська'],
       'ur' => ['Urdu', /* Left-to-right marker "‭" */ 'اردو', LanguageInterface::DIRECTION_RTL],
