diff --git a/src/Notification.php b/src/Notification.php
--- a/src/Notification.php
+++ b/src/Notification.php
@@ -2,7 +2,6 @@
 
 namespace Drupal\content_moderation_notifications;
 
-use Drupal\Component\Utility\DeprecationHelper;
 use Drupal\Core\Config\ConfigFactoryInterface;
 use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
@@ -78,12 +77,12 @@
         '#template' => $subject,
         '#context' => $data['params']['context'],
       ];
-      $subject = DeprecationHelper::backwardsCompatibleCall(
-        currentVersion: \Drupal::VERSION,
-        deprecatedVersion: '10.3',
-        currentCallable: fn() => $this->renderer->renderInIsolation($template),
-        deprecatedCallable: fn() => $this->renderer->renderPlain($template),
-      );
+      if (method_exists($this->renderer, 'renderInIsolation')) {
+        $subject = $this->renderer->renderInIsolation($template);
+      }
+      else {
+        $subject = $this->renderer->renderPlain($template);
+      }
       // Remove any newlines from Subject.
       $subject = trim(str_replace("\n", ' ', $subject));
       $data['params']['subject'] = $subject;
@@ -95,12 +94,12 @@
         '#template' => $message,
         '#context' => $data['params']['context'],
       ];
-      $message = DeprecationHelper::backwardsCompatibleCall(
-        currentVersion: \Drupal::VERSION,
-        deprecatedVersion: '10.3',
-        currentCallable: fn() => $this->renderer->renderInIsolation($template),
-        deprecatedCallable: fn() => $this->renderer->renderPlain($template),
-      );
+      if (method_exists($this->renderer, 'renderInIsolation')) {
+        $message = $this->renderer->renderInIsolation($template);
+      }
+      else {
+        $message = $this->renderer->renderPlain($template);
+      }
       $data['params']['message'] = check_markup($message, $notification->getMessageFormat());
 
       // Figure out who the email should be going to.
@@ -148,12 +147,12 @@
         '#template' => $adhoc_emails,
         '#context' => $data['params']['context'],
       ];
-      $adhoc_emails = DeprecationHelper::backwardsCompatibleCall(
-        currentVersion: \Drupal::VERSION,
-        deprecatedVersion: '10.3',
-        currentCallable: fn() => $this->renderer->renderInIsolation($template),
-        deprecatedCallable: fn() => $this->renderer->renderPlain($template),
-      );
+      if (method_exists($this->renderer, 'renderInIsolation')) {
+        $adhoc_emails = $this->renderer->renderInIsolation($template);
+      }
+      else {
+        $adhoc_emails = $this->renderer->renderPlain($template);
+      }
 
       // Split Adhoc emails on commas and newlines.
       $adhoc_emails = array_map('trim', explode(',', preg_replace("/((\r?\n)|(\r\n?))/", ',', $adhoc_emails)));
