diff --git a/entity_usage.views.inc b/entity_usage.views.inc
index afeadd9..393c19d 100644
--- a/entity_usage.views.inc
+++ b/entity_usage.views.inc
@@ -31,6 +31,13 @@ function entity_usage_views_data() {
     ],
   ];
 
+  $data['entity_usage']['source_id'] = [
+    'title' => t('Usage source ID'),
+    'help' => t('The source ID, e.g. nid of the using content. Select COUNT(DISTINCT) in the Aggregation Settings of this field to display the number of unique source IDs'),
+    'field' => [
+      'id' => 'numeric',
+    ],
+  ];
   return $data;
 }
 
