diff --git a/core/lib/Drupal/Core/Entity/EntityStorageBase.php b/core/lib/Drupal/Core/Entity/EntityStorageBase.php
index 0182b75..cad2acd 100644
--- a/core/lib/Drupal/Core/Entity/EntityStorageBase.php
+++ b/core/lib/Drupal/Core/Entity/EntityStorageBase.php
@@ -112,6 +112,9 @@ abstract class EntityStorageBase extends EntityHandlerBase implements EntityStor
    * {@inheritdoc}
    */
   public function getEntityClass(?string $bundle = NULL): string {
+    if (empty($this->baseEntityClass)) {
+      $this->baseEntityClass = $this->entityType->getClass();
+    }
     return $this->baseEntityClass;
   }
 
